cluster/secrets.go

```diff
@@ -164,25 +165,43 @@ func InstallSecretByK8SConfig(kubeConfig []byte, orgID uint, secretName string,
 		return nil, errors.Wrap(err, "failed to create kubernetes client")
 	}
 
-	secretItem, err := secret.Store.GetByName(orgID, req.SourceSecretName)
-	if err == secret.ErrSecretNotExists {
-		return nil, ErrSecretNotFound
-	} else if err != nil {
-		return nil, emperror.With(errors.Wrap(err, "failed to get secret"), "secret", req.SourceSecretName)
-	}
-
 	kubeSecretRequest := intSecret.KubeSecretRequest{
 		Name:      secretName,
 		Namespace: req.Namespace,
-		Type:      secretItem.Type,
-		Values:    secretItem.Values,
+		Type:      "",
+		Values:    nil,
 		Spec:      intSecret.KubeSecretSpec{},
 	}
 
+	sourceMeta := secretTypes.K8SSourceMeta{
+		Name:     secretName,
+		Sourcing: secretTypes.EnvVar,
+	}
+
+	if req.SourceSecretName != "" {
+		secretItem, err := secret.Store.GetByName(orgID, req.SourceSecretName)
+		if err == secret.ErrSecretNotExists {
+			return nil, ErrSecretNotFound
+		} else if err != nil {
+			return nil, emperror.With(errors.Wrap(err, "failed to get secret"), "secret", req.SourceSecretName)
+		}
+
+		kubeSecretRequest = intSecret.KubeSecretRequest{
+			Name:      secretName,
+			Namespace: req.Namespace,
+			Type:      secretItem.Type,
+			Values:    secretItem.Values,
+			Spec:      intSecret.KubeSecretSpec{},
```

Comment:
We know the size of the map, right (`req.Spec`)? Maybe initialize it first time the struct is created?